src/project-tests/technical-docs-tests.js

```diff
@@ -59,7 +59,8 @@ export default function createTechnicalDocsPageTests() {
                     
                     const headerText = mainSections.map((el, i) => {
                       FCC_Global.assert.isAbove(el.firstElementChild.innerText.length, 0, 'All headers must contain text ')
-                      return [][i] = el.firstElementChild.innerText.replace(/\s/g, '_');
+                      //remove leading / trailing spaces in headerText and then replace all other spaces with underscores.
+                      return [][i] = el.firstElementChild.innerText.replace(/^\s+|\s+$/g, '').replace(/\s/g, '_');
```

Comment:
@tbushman ok, so a few things here:

- **first**: ultimately, I think the solution using `trim` is the best solution since it's more succinct and this first `replace` regex does the same thing. No need to use that when we have the built in function. I was just providing the regex by way of example to show that there was another way besides chaining the `replace` method. 

Let's change this to:
```js
el.firstElementChild.innerText.trim().replace(/\s/g, '_');
```

- **second**: what was the deal with replacing the elipsis? Not sure why we were doing this, but if there was a good reason to, we should still be doing it, but just accomplishing it in one `replace` method: e.g. `.replace(/\s|\.\.\./g, '_')`

- **finally** - it would seem that I introduced some poor code here. Returning an assignment from a function as I was doing here is poor practice, even though in this case it works. Let's do this instead:

```
let arr = [][i] = el.firstElementChild.innerText.replace(/^\s+|\s+$/g, '').replace(/\s/g, '_');
return arr;
```
for both this one and the function right below it. Feel free to change the assignment if we are going back to replacing the elipsis (though again, not sure where that came from in the first place).

Does this all make sense? If you need help on anything, feel free to hit me up on Gitter! Happy to help.

Thanks!